exporters/otlp/internal/envconfig/envconfig.go

```diff
@@ -90,19 +100,42 @@ func WithURL(n string, fn func(*url.URL)) func(e *EnvOptionsReader) {
 	}
 }
 
-// WithTLSConfig retrieves the specified config and passes it to ConfigFn as a crypto/tls.Config.
-func WithTLSConfig(n string, fn func(*tls.Config)) func(e *EnvOptionsReader) {
+// WithCertPool retrieves the specified config and passes it to ConfigFn as a crypto/x509.CertPool
+// constructed from reading the certificate at the given path.
+func WithCertPool(n string, fn func(*x509.CertPool)) func(e *EnvOptionsReader) {
 	return func(e *EnvOptionsReader) {
 		if v, ok := e.GetEnvValue(n); ok {
 			if b, err := e.ReadFile(v); err == nil {
-				if c, err := createTLSConfig(b); err == nil {
+				if c, err := createCertPool(b); err == nil {
 					fn(c)
 				}
 			}
 		}
 	}
 }
 
+// WithCertPool retrieves the specified configs and passes it to ConfigFn as a crypto/tls.Certificate
+// constructed from reading the key pair at the given paths. Both files must exist.
+func WithClientCert(nc, nk string, fn func(tls.Certificate)) func(e *EnvOptionsReader) {
+	return func(e *EnvOptionsReader) {
+		vc, okc := e.GetEnvValue(nc)
+		vk, okk := e.GetEnvValue(nk)
+		if !okc || !okk {
+			return
+		}
+		cert, errc := e.ReadFile(vc)
+		key, errk := e.ReadFile(vk)
+		if errc != nil && errk != nil {
```

Comment:
We should likely log these errors so the user known why their connection doesn't have TLS setup correctly: 

https://github.com/open-telemetry/opentelemetry-go/blob/e8023fab22dc1cf95b47dafcc8ac8110c6e72da1/internal/global/internal_logging.go#L52

Reply:
Done, also added logging for other silently ignored errors in the file